[jugheadjones10] iP

jugheadjones10 opened the pull request.

Commit:
Add Gradle support

Commit:
Level-1 changes

Commit:
Do Level 2

Commit:
Do Level 3

Commit:
Do level 4

Commit:
A-TextUiTesting

Commit:
Level 5

Commit:
Level 6

Commit:
Add changes for Level 7

Commit:
Do level 8

Commit:
Merge remote-tracking branch 'origin/branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java

Commit:
Fixed level 8 bug

Commit:
Do A-MoreOOP

Commit:
Fix some bugs

Commit:
Do A-Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

markusyeo commented on src/main/java/yj/Event.java:
```diff
@@ -0,0 +1,44 @@
+    @Override
+    public String toSaveString() {
+        // E | 0 | project meeting | Aug 6th 2-4pm
+        return "E | " + (isDone ? 1 : 0) + " | " + super.toString() + " | " + from.format(dateTimeFormatter) + " | " + to.format(dateTimeFormatter);
```
This line of code exceeds 100 characters, you might want to split the code here to improve code readability

markusyeo commented on src/main/java/yj/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        return result;
+    }
+
+
```
You might want to perhaps consider deleting the additional whitespace lines

markusyeo commented on src/main/java/yj/Storage.java:
```diff
@@ -0,0 +1,79 @@
+                String description = taskLineParts[2].trim();
+
+                Task task;
+                switch (taskType) {
```
You might want to following the coding standard here to not indent cases in switch statements

markusyeo reviewed (commented):
You can consider adding a checkstyle as suggested in Week 4's tasks to help eliminate coding style violations

roultitude commented on src/main/java/yj/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        return description;
+    }
+
+    static public Map<String, String> parseDeadlineCommand(String input) {
+        String by = input.split("/by")[1].trim();
+        String description = input.split("/by")[0].trim();
+        Map<String, String> result = new HashMap<String, String>();
+        result.put("description", description);
+        result.put("by", by);
+        return result;
+    }
+
+    static public Map<String, String> parseEventCommand(String input) {
+        String from = input.split("/from")[1].split("/to")[0].trim();
+        String to = input.split("/to")[1].trim();
+        String description = input.split("/from")[0].trim();
+
+        Map<String, String> result = new HashMap<String, String>();
+        result.put("description", description);
+        result.put("from", from);
+        result.put("to", to);
+        return result;
+    }
```
Interesting use of HashMaps, but might be adding complexity where otherwise not required. Also results in use of string keys that are hardcoded into the main program below (YJ.java).

roultitude commented on src/main/java/yj/YJ.java:
```diff
@@ -0,0 +1,124 @@
+                case EVENT:
+                    try {
+                        Map<String, String> result = Parser.parseEventCommand(input);
+                        Event newEvent = new Event(result.get("description"), result.get("from"), result.get("to"));
```
Kind of dangerous to be using string keys here, maybe code in as variables for the Event class and implement getters and setters to throw compile time errors if any variable name is changed later on?

roultitude commented on src/main/java/yj/YJ.java:
```diff
@@ -0,0 +1,124 @@
+
+        while (true) {
+            Command command = Parser.parseCommand(input);
+            switch (command) {
+                case LIST:
+                    List<Task> tasksList = tasks.getTasks();
+                    for (int i = 0; i < tasksList.size(); i++) {
+                        Task task = tasksList.get(i);
+                        ui.print((i + 1) + "." + task.toString());
+                    }
```
Switch cases can be left unindented as per the style guide for readability.

roultitude reviewed (commented):
Structure of the program could be improved, i.e. most of the logic is still within the main program body. Perhaps consider moving the logic into the parser class instead?

Commit:
More gradle changes

Commit:
A-JUnit changes

Commit:
Added Javadocs

Commit:
Fixed some SLAP

Commit:
Level 9

Commit:
Revert "Level 9"

This reverts commit cd786e177ef26c67b5843b20380110b2bed9deaa.

Commit:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

Commit:
Added JavaFX integration

Commit:
Refine UI to use FX

Commit:
Finalized UI

Commit:
Remove text UI related stuff

Commit:
Use assert

Commit:
Made code quality improvements like less code nesting

jugheadjones10 committed:
Merge pull request #2 from jugheadjones10/branch-A-Assertions

Use assert

Commit:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

jugheadjones10 committed:
Merge pull request #3 from jugheadjones10/branch-A-CodeQuality

Made code quality improvements like less code nesting

Commit:
Allow friendlier commands in the style of linux commands

Commit:
Update the docs README to include explanations on all features

Added explanations on commands such as list, delete, todo, deadline,
event, etc.

For each command, included the command that the user should type into
the chatbot as well as the output that the user will see as a result.

Commit:
Improve file processing to work with jar
